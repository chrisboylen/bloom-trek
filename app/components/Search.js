import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import { SearchBar } from 'react-native-elements';
import PropTypes from 'prop-types';

export default class Search extends Component {
  constructor() {
    super();
    this.state = {
      searchInput: '',
    };
  }

  handleCancel = () => {
    this.setState({
      searchInput: '',
    });
  };

  handleChangeText = (input) => {
    this.setState({
      searchInput: input,
    });
  };

  handleSubmit = () => {
    const { updateLocation } = this.props;
    const { searchInput } = this.state;

    updateLocation(searchInput);
  }

  render() {
    const { searchInput } = this.state;

    return (
      <SearchBar
        round
        containerStyle={styles.search}
        inputStyle={styles.searchInput}
        clearIcon={{ color: '#005900' }}
        showLoading
        searchIcon
        onChangeText={input => this.setState({ searchInput: input })}
        onCancel={this.handleCancel}
        value={searchInput}
        placeholder="search location"
        onSubmitEditing={this.handleSubmit}
      />
    );
  }
}

const styles = StyleSheet.create({
  search: {
    backgroundColor: '#F9F5ED',
    borderBottomWidth: 0,
    borderTopWidth: 0,
  },
  searchInput: {
    backgroundColor: '#c1e6b1',
    color: '#005900',
  },
});

Search.propTypes = {
  updateLocation: PropTypes.func,
};
