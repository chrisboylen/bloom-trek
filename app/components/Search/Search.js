import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import { SearchBar } from 'react-native-elements';
import PropTypes from 'prop-types';

export default class Search extends Component {
  constructor() {
    super();
    this.state = {
      searchInput: '',
    };
  }

  render() {
    const { searchInput } = this.state;
    const { updateLocation } = this.props;

    return (
      <SearchBar
        style={styles.search}
        showLoading
        clearIcon={{ color: 'red' }}
        searchIcon
        onChangeText={input => this.setState({ searchInput: input })}
        onCancel={() => this.setState({ searchInput: '' })}
        value={searchInput}
        placeholder="Search location"
        onSubmitEditing={() => updateLocation(searchInput)}
      />
    );
  }
}

const styles = StyleSheet.create({
  search: {
    flex: 1,
    top: 100,
  },
});

Search.propTypes = {
  updateLocation: PropTypes.func,
};
