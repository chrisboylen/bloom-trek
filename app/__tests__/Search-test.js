import React from 'react';
import { shallow } from 'enzyme';
import Search from '../components/Search/Search';

describe('Search', () => {
  let wrapper;
  let mockUpdateLocation;

  beforeEach(() => {
    mockUpdateLocation = jest.fn();
    wrapper = shallow(
      <Search updateLocation={mockUpdateLocation} />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should set state when input is changed', () => {
    wrapper.setState({ searchInput: 'aspen' });

    expect(wrapper.state().searchInput).toEqual('aspen');
  });

  describe('handleCancel', () => {
    it('should set state when invoked', () => {
      wrapper.setState({ searchInput: 'aspen' });
      expect(wrapper.state().searchInput).toBe('aspen');

      wrapper.instance().handleCancel();

      expect(wrapper.state().searchInput).toBe('');
    });
  });

  describe('handleChangeText', () => {
    it('should set state when invoked', () => {
      const input = 'boulder';
      wrapper.instance().handleChangeText(input);

      expect(wrapper.state().searchInput).toBe('boulder');
    });
  });

  describe('handleSubmit', () => {
    it('should invoke updateLocation when called', () => {
      wrapper.setState({ searchInput: 'aspen' });
      wrapper.instance().handleSubmit();

      expect(mockUpdateLocation).toHaveBeenCalled();
    });
  });
});
