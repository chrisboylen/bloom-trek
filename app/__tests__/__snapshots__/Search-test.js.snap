// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    updateLocation={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "clearIcon": Object {
        "color": "#005900",
      },
      "containerStyle": Object {
        "backgroundColor": "#F9F5ED",
        "borderBottomWidth": 0,
        "borderTopWidth": 0,
      },
      "icon": Object {},
      "inputStyle": Object {
        "backgroundColor": "#c1e6b1",
        "color": "#005900",
      },
      "lightTheme": false,
      "loadingIcon": Object {},
      "noIcon": false,
      "onCancel": [Function],
      "onChangeText": [Function],
      "onSubmitEditing": [Function],
      "placeholder": "search location",
      "placeholderTextColor": "#86939e",
      "round": true,
      "searchIcon": true,
      "showLoading": true,
      "showLoadingIcon": false,
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "clearIcon": Object {
          "color": "#005900",
        },
        "containerStyle": Object {
          "backgroundColor": "#F9F5ED",
          "borderBottomWidth": 0,
          "borderTopWidth": 0,
        },
        "icon": Object {},
        "inputStyle": Object {
          "backgroundColor": "#c1e6b1",
          "color": "#005900",
        },
        "lightTheme": false,
        "loadingIcon": Object {},
        "noIcon": false,
        "onCancel": [Function],
        "onChangeText": [Function],
        "onSubmitEditing": [Function],
        "placeholder": "search location",
        "placeholderTextColor": "#86939e",
        "round": true,
        "searchIcon": true,
        "showLoading": true,
        "showLoadingIcon": false,
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
